src/nvim/os/env.c

```diff
@@ -169,6 +178,64 @@ void os_get_hostname(char *hostname, size_t size)
 #endif
 }
 
+/// @return -1 if error encountered, 1 if capslock active otherwise 0 if not.
```

Comment:
Would rather suggest to use boolean and `const char **const error` argument (`*error == NULL` if not error). This lacks error reporting capabilities.

Reply:
Ok sure I will do it